natsort/compat/locale.py

```diff
@@ -38,10 +38,10 @@ def dumb_sort() -> bool:
 
     # If using icu, get the locale from the current global locale,
     def get_icu_locale() -> str:
-        try:
-            return cast(str, icu.Locale(".".join(getlocale())))
-        except TypeError:  # pragma: no cover
+        language_code, encoding = getlocale()
+        if language_code is None or encoding is None:  # pragma: no cover
             return cast(str, icu.Locale())
+        return icu.Locale(f"{language_code}.{encoding}")
```

Comment:
This is still failing. I think it is because `icu` has no associated types so the default that `mypy` assumes is `Any`. That's the reasons for the `cast` to `str.`

That should fix it.